test-app/src/main/java/org/readium/r2/testapp/Application.kt

```diff
@@ -79,14 +80,22 @@ class Application : android.app.Application() {
                 readium.publicationOpener,
                 readium.assetRetriever,
                 createPublicationRetriever = { listener ->
+                    val publicationImporter = PublicationImporter(
+                        listener = listener,
+                        storageDir = storageDir,
+                        assetRetriever = readium.assetRetriever
+                    )
+
                     PublicationRetriever(
                         listener = listener,
+                        publicationImporter = publicationImporter,
```

Comment:
I think you can simplify a bit the `PublicationImporter` and having it only as a dependency of the `LocalPublicationRetriever`.
As in `PublicationImporter`, it's only used in the `LocalPublicationRetriever`'s listener.

Reply:
You can try this patch I made for what I had in mind: [simplify.diff.txt](https://github.com/readium/kotlin-toolkit/files/14169264/simplify.diff.txt). I think it could be simplified further actually, I'm not sure why we need the `PublicationImporter` instead of a private function in `LocalPublicationRetriever` here.

I couldn't check if I broke something though, because both OPDS and LCP downloads were already broken in the PR.

Reply:
What's wrong with OPDS and LCP? They perfectly work on my device.

Reply:
I get these errors.

![Screenshot_1707155725](https://github.com/readium/kotlin-toolkit/assets/58686775/d5eb9cd8-ca6f-4eec-9fdb-1b51627117d6)
![Screenshot_1707155738](https://github.com/readium/kotlin-toolkit/assets/58686775/0b3570f4-0649-4e77-9752-908ba5b0e39d)

Reply:
I just double-checked, and it works from `develop`.

Reply:
It's wrong because the contract is that `PublicatinRetriever` always return a file in the the library dir. That's exactly the purpose of `PublicationImporter`.